### DISCUSSION TITLE
Ability to prompt when lack of existing .chezmoidata

### DISCUSSION THREAD

**Discussion by Roxedus** [twpayne/chezmoi#3449] [General]
I have usecases for both using the prompt functions, and preseeding `chezmoi.EXT`.

While the prompt works (when setting a default value outside of an if), it discards any existing `chezmoi.EXT` already present on the machine. This makes it impossible to use the same configuration on automated setups (ie placing `chezmoi.EXT` before running init) and manual setups where the prompts would answer all the needed questions needed for templating. 

I understand this will break the current expected functionality. 

```sh
>pwd
/config/.local/share/chezmoi
>yq '.data.name' ~/.config/chezmoi/chezmoi.yaml
Roxedus
>yq '.name' .chezmoidata.yaml
Roxedus
>cat .chezmoi.yaml.tmpl
{{- $personal = promptBool "Personal?" }}
data:
  email: {{ $personal | quote }}
  name: {{ $name }}

>chezmoi init roxedus --dry-run
chezmoi: template: chezmoi.yaml:4: undefined variable "$name"
```

I found #3225 and understand that *an* solution would be to add lots of logic to the template for tty and whatnot.

**Comment by halostatue** [comment]
This would be the `prompt*Once` init functions. Your `.chezmoi.yaml.tmpl` has issues and should be (I’m using left delimiter specification because it does not conflict with YAML formatters):

```gotemplate
# chezmoi:template:left-delimiter="(:" right-delimiter=":)"
(:- $personal := promptBoolOnce . "email" "Personal?" -:)
(:- $name := promptStringOnce . "name" "Name" -:)
data:
  email: (: $personal | quote :)
  name: (: $name | quote :)
```

**Reply by Roxedus** [reply]
Thanks, the *once prompts does what i need. The docs isn't very clear on when it uses the existing data context, which is what threw me off that function altogether.

**Reply by halostatue** [reply]
The `.` is the existing data context.
